artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/proton/plug/ActiveMQProtonConnectionCallback.java

```diff
@@ -85,7 +105,42 @@ public boolean isSupportsAnonymous() {
    }
 
    @Override
+   public void init() throws Exception {
+      //This internal core session is used to represent the connection
+      //in core server. It is used to identify unique clientIDs.
+      //Note the Qpid-JMS client does create a initial session
+      //for each connection. However is comes in as a Begin
+      //After Open. This makes it unusable for this purpose
+      //as we need to decide the uniqueness in response to
+      //Open, and the checking Uniqueness and adding the unique
+      //client-id to server need to be atomic.
+      if (internalSession == null) {
+         SASLResult saslResult = amqpConnection.getSASLResult();
+         String user = null;
+         String passcode = null;
+         if (saslResult != null) {
+            user = saslResult.getUser();
+            if (saslResult instanceof PlainSASLResult) {
+               passcode = ((PlainSASLResult) saslResult).getPassword();
+            }
+         }
+         internalSession = manager.getServer().createSession(createInternalSessionName(), user, passcode, ActiveMQClient.DEFAULT_MIN_LARGE_MESSAGE_SIZE, protonConnectionDelegate, // RemotingConnection remotingConnection,
```

Comment:
I would rather refactor ActiveMQServer to handle this instead of creating an internalSession. It gets over-copmlicated for something this simple.


We can talk offline on how we would do such refactoring.